### PR TITLE
Fix empty queries in sqlite testrunner

### LINKED ISSUE #343: Two queries from SQLToResultTest result in an empty table
After migrating the tests from SQLToResultTest to SQLiteTestRunner I stumbled upon two tests performed on TPC-H tables which result in empty tables, is this intended?

`SELECT customer.c_custkey, customer.c_name, COUNT(orders.o_orderkey)
        FROM customer JOIN orders ON c_custkey = o_custkey
        GROUP BY customer.c_custkey, customer.c_name
        HAVING COUNT(orders.o_orderkey) >= 100;`
and
`SELECT customer.c_custkey, customer.c_name, COUNT(orderitems.o_orderkey)
       FROM customer JOIN (
         SELECT * FROM orders JOIN lineitem ON o_orderkey = l_orderkey
       ) AS orderitems
       ON customer.c_custkey = orderitems.o_custkey
       GROUP BY customer.c_custkey, customer.c_name
       HAVING COUNT(orderitems.o_orderkey) >= 100;`

### PR DESCRIPTION
**Pull request by mrks**
fixes #343

### CODE REVIEW

**Comment by lanice** [comment]
It looks like SQLite is presenting the column name(s) a little different than hyrise in the failing query?

`Got: c_custkey (int)`
`Expected: customer.c_custkey (int)`

File: src/test/testing_assert.cpp [hunk 1]
```diff
@@ -118,42 +118,52 @@ ::testing::AssertionResult check_table_equal(const Table& tleft, const Table& tr
     std::sort(right.begin(), right.end());
   }
 
+  bool has_failed = false;
   for (unsigned row = 0; row < left.size(); row++)
     for (ColumnID col{0}; col < left[row].size(); col++) {
       if (is_null(left[row][col]) || is_null(right[row][col])) {
         EXPECT_TRUE(is_null(left[row][col]) && is_null(right[row][col]));
+        has_failed = true;
```
Comment (lanice) [inline src/test/testing_assert.cpp]: I think this causes it to fail EVERY time. We'd have to basically do our 'own' `EXPECT_TRUE`, `EXPECT_EQ`, etc., which sets the has failed. In this case here for example:

`if (! ( is_null(left[row][col]) && is_null(right[row][col]) ) ) { has_failed = true; }`

**Comment by mrks** [comment]
Yes, that was a stupid quick shot :D

**Comment by mrks** [comment]
@lanice Not sure if you get notified if I dismiss a review - could you have another look?

**Comment by lanice** [comment]
👍

**Pull request merged**